lib/crawler.js

```diff
@@ -230,42 +237,222 @@ util.inherits(Crawler, EventEmitter);
+        agent: agent,
+        headers: {
+            "User-Agent": crawler.userAgent,
+            "Host": queueItem.host + (queueItem.port !== 80 ? ":" + queueItem.port : "")
+        }
+    };
+
+    if (queueItem.referrer) {
+        requestOptions.headers.Referer = queueItem.referrer;
+    }
+
+    // If port is one of the HTTP/HTTPS defaults, delete the option to avoid conflicts
+    if (requestOptions.port === 80 || requestOptions.port === 443) {
+        delete requestOptions.port;
+    }
+
+    // Add cookie header from cookie jar if we're configured to
+    // send/accept cookies
+    if (crawler.acceptCookies && crawler.cookies.getAsHeader()) {
+        requestOptions.headers.cookie =
+            crawler.cookies.getAsHeader(queueItem.host, queueItem.path);
+    }
+
+    // Add auth headers if we need them
+    if (crawler.needsAuth) {
+        var auth = crawler.authUser + ":" + crawler.authPass;
+
+        // Generate auth header
+        auth = "Basic " + new Buffer(auth).toString("base64");
+        requestOptions.headers.Authorization = auth;
+    }
+
+    // Add proxy auth if we need it
+    if (crawler.proxyUser !== null && crawler.proxyPass !== null) {
+        var proxyAuth = crawler.proxyUser + ":" + crawler.proxyPass;
+
+        // Generate auth header
+        proxyAuth = "Basic " + new Buffer(proxyAuth).toString("base64");
+        requestOptions.headers["Proxy-Authorization"] = proxyAuth;
+    }
+
+    // And if we've got any custom headers available
+    if (crawler.customHeaders) {
+        for (var header in crawler.customHeaders) {
+            if (!crawler.customHeaders.hasOwnProperty(header)) {
+                continue;
+            }
+
+            requestOptions.headers[header] = crawler.customHeaders[header];
+        }
+    }
+
+    return requestOptions;
+};
+
+/*
+    Public: Initiates a request for the robots.txt resource on crawler's target
+    domain.
+
+    callback - Callback with signature: error, robotsTxtUrl, responseBuffer. All arguments are strings.
+
+    Returns the crawler object, to enable chaining.
+
+*/
+Crawler.prototype.getRobotsTxt = function (callback) {
+    var crawler = this;
+
+    var client = crawler.initialProtocol === "https" ? https : http;
+
+    // Apply the ignoreInvalidSSL setting to https connections
+    if (client === https && crawler.ignoreInvalidSSL === true) {
+        client.rejectUnauthorized = false;
+        client.strictSSL = false;
+    }
+
+    var robotsTxtUrl = uri({
+        protocol: crawler.initialProtocol,
+        hostname: crawler.host,
+        port: crawler.initialPort,
+        path: "/robots.txt"
+    }).href();
+
+    var requestOptions = crawler.getRequestOptions({
+        protocol: crawler.initialProtocol,
+        host: crawler.host,
+        port: crawler.initialPort,
+        path: "/robots.txt",
+        url: robotsTxtUrl
+    });
+
+    // Get the resource!
+    var clientRequest = client.request(requestOptions, function(response) {
```

Comment:
Yeah, I know - I think the main culprit here is the fact that the `getRequestOptions` logic expects queueItem.url to be defined as well as all the individual parts of the URL. Maybe I could change it so that if only a URL is provided to getRequestOptions, it knows how to construct a `http.request` options object from just that. That helps us avoid some repetition here